tsec-libsodium/src/main/scala/tsec/libsodium/kx/kx.scala

```diff
@@ -0,0 +1,42 @@
+package tsec.libsodium
+
+import cats.evidence.Is
+import tsec.common.HKByteArrayNewt
+
+package object kx {
+
+  private[tsec] val SodiumKey$$ : HKByteArrayNewt = new HKByteArrayNewt {
+    type Repr[A] = Array[Byte]
+
+    def is[G] = Is.refl[Array[Byte]]
+  }
+
+  type SodiumKey[A] = SodiumKey$$.Repr[A]
+
+  object SodiumKey {
+    def apply[A](bytes: Array[Byte]): SodiumKey[A]   = is[A].coerce(bytes)
+    @inline def is[A]: Is[Array[Byte], SodiumKey[A]] = SodiumKey$$.is[A]
+  }
+
+
+  trait PublicKey
+  trait SecretKey
+  trait SharedKey
```

Comment:
We do not need these. `SecretKey[KeyExchange]` or `PublicKey[CryptoBox]` or whatever is enough.